modules/git/repo_commitgraph.go

```diff
@@ -0,0 +1,24 @@
+package git
+
+import (
+	"os"
+	"path"
+
+	"gopkg.in/src-d/go-git.v4/plumbing/format/commitgraph"
+	cgobject "gopkg.in/src-d/go-git.v4/plumbing/object/commitgraph"
+)
+
+// CommitNodeIndex returns the index for walking commit graph
+func (r *Repository) CommitNodeIndex() (cgobject.CommitNodeIndex, *os.File) {
+	indexPath := path.Join(r.Path, "objects", "info", "commit-graph")
+
+	file, err := os.Open(indexPath)
+	if err == nil {
+		index, err := commitgraph.OpenFileIndex(file)
+		if err == nil {
+			return cgobject.NewGraphCommitNodeIndex(index, r.gogitRepo.Storer), file
+		}
+	}
+
```

Comment:
If err is not file not exist error, we should return it but not hide it.

Reply:
I don't think we should return it. This is quite graceful error handling to the regular git object storage. Worst case is a performance penalty but it will still yield the same results.

Reply:
OK. but at least, we should know commit graph is not using since something wrong. We need an error log or warning log if commitgraph file exist but load failed. That will be helpful when we investigate errors.

Reply:
@lunny is my change  to lines 30-33 acceptable?